pkg/filters/string.go

```diff
@@ -220,78 +217,61 @@ func (f *StringFilter) FilterOut() bool {
 	return true
 }
 
+func (f *StringFilter) DeepCopy() *StringFilter {
+	if f == nil {
+		return nil
+	}
+
+	n := &StringFilter{}
+
+	maps.Copy(n.equal, f.equal)
+	maps.Copy(n.notEqual, f.notEqual)
+	f.prefixes.DeepCopy()
+	n.prefixes = *f.prefixes.DeepCopy()
+	n.suffixes = *f.suffixes.DeepCopy()
+	maps.Copy(n.contains, f.contains)
+	n.notPrefixes = *f.notPrefixes.DeepCopy()
+	n.notSuffixes = *f.notSuffixes.DeepCopy()
+	maps.Copy(n.notContains, f.notContains)
+	n.enabled = f.enabled
+
+	return n
+}
+
 type BPFStringFilter struct {
 	StringFilter
-	mapName string
 }
```

Comment:
> Is there a reason to keep the BPFStringFilter if we remove the mapName from it?

No there's not.

> We can use StringFilter instead, no?

Yeah, I had pondered about this and forgot to put TODO comments, since the BPFFilters won't be for BPF anymore. They will only provide the equality values.

Reply:
`// TODO:` added.